app/test_main.py

```diff
@@ -1 +1,39 @@
-# write your code here
+from unittest import mock
+
+
+from app.main import can_access_google_page
+
+
+class TestAccessPage:
+    def test_access_google_page_with_invalid_url(self) -> None:
+        with mock.patch("app.main.valid_google_url")\
+             as mock_valid_google_url,\
+             mock.patch("app.main.has_internet_connection")\
```

Comment:
you have three tests that do the same thing and the only difference is return values of mocked functions. Would make more sense to use parametrize here